### PR TITLE
Query Param filterBy

### LINKED ISSUE #1062: Permitir filtro pelo nome do model
## Feature Description

Para resolvermos o problema dos selects que não exibem todos os itens, vamos usar a busca assíncrona na API. Para isso, esses endpoints precisam permitir um filtro pelo nome do model, passando um query param (`modelName`, por exemplo). A busca será feita pelos itens que comecem com a string passada (`like string%`)

Atualmente acredito que precisamos no endpoint de organizações e usuários

---------------

## Acceptance criteria

* Conseguir pesquisar pelo nome do model passando um query param

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by alexandreadames**
## Summary

Resolves #1062 

## Relevant technical choices

In all endpoints, was added the query param: `filterBy` and `filter`, ex

`GET institutions?filterBy=name&filter=pw`

`filterBy`: Collumm in database
`filter`: value to search

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code passes the linting.
- [ ] My code has proper inline documentation.
- [ ] I have manually tested this PR.

### CODE REVIEW

**Review by mateus4k — DISMISSED** [review]

**Comment by mateus4k** [comment]
Se puder cobrir por testes acho que seria uma boa também

File: packages/api/app/Models/Traits/Params.js [hunk 1]
```diff
@@ -199,14 +199,35 @@ class Params {
...
+				filter,
+			} = request.params;
 
 			// eslint-disable-next-line no-underscore-dangle
 			const resource = this._single.table;
 
 			const { filterById = true, skipRelationships = [], skipPagination = false } = options;
 
+			if (filterBy && filter) {
+				this.where(filterBy, 'LIKE', `${filter}%`);
```
Comment (luizeboli) [inline packages/api/app/Models/Traits/Params.js:228]:
```suggestion
				this.where(filterBy, 'LIKE', `%${filter}%`);
```

**Review by mateus4k — APPROVED** [review]

**Review by luizeboli — APPROVED** [review]

File: packages/api/app/Models/Traits/Params.js [hunk 2]
```diff
@@ -199,14 +199,35 @@ class Params {
 			],
 		};
 
+		/* const collumNames = {	
+			institutions: 'name',
+			services: 'name',
+			users: 'first_name',
+		}; */
```
Comment (luizeboli) [inline packages/api/app/Models/Traits/Params.js:206]:
```suggestion
```

**Review by luizeboli — CHANGES_REQUESTED** [review]

File: packages/api/app/Models/Traits/Params.js [hunk 3]
```diff
@@ -199,14 +199,35 @@ class Params {
...
+
```
Comment (luizeboli) [inline packages/api/app/Models/Traits/Params.js:207]:
```suggestion
```

**Review by luizeboli — CHANGES_REQUESTED** [review]

**Review by luizeboli — DISMISSED** [review]

**Review by mateus4k — DISMISSED** [review]

**Review by mateus4k — APPROVED** [review]
Nice!

**Pull request merged**